ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -476,59 +459,7 @@ module Intel = struct
           in
           vendor_name, device))
     in
-    let bar_size =
-      List.nth device.Pci.Pci_dev.size 2
-      |> Int64.of_nativeint
-    in
-    let rec collect acc = function
-      | []             -> acc
-      | Some v :: tail -> collect (v :: acc) tail
-      | None   :: tail -> collect acc tail
-    in
-    whitelist
-    |> List.map
-      Identifier.(fun conf ->
-          let vgpus_per_pgpu =
-            bar_size /// 1024L /// 1024L
-            /// conf.identifier.low_gm_sz
-            --- 1L
-          in
-          if vgpus_per_pgpu <= 0L then 
-            begin
-              warn "Not enough memory for Intel VGPUs. \
-                    If you intend to use them, increase the GPU \
-                    BAR size in the BIOS settings.";
-              None
-            end
-          else
-            let vgpu_size =
-              Constants.pgpu_default_size /// vgpus_per_pgpu
-            in
-            let internal_config = let open Xapi_globs in
-              List.concat [
-                [ vgt_low_gm_sz, Int64.to_string conf.identifier.low_gm_sz
-                ; vgt_high_gm_sz, Int64.to_string conf.identifier.high_gm_sz
-                ; vgt_fence_sz, Int64.to_string conf.identifier.fence_sz
-                ]
-              ; match conf.identifier.monitor_config_file with
-                | Some monitor_config_file ->
-                  [vgt_monitor_config_file, monitor_config_file]
-                | None -> []
-              ]
-            in
-            Some {
-              vendor_name;
-              model_name = conf.model_name;
-              framebuffer_size = conf.framebufferlength;
-              max_heads = conf.num_heads;
-              max_resolution_x = conf.max_x;
-              max_resolution_y = conf.max_y;
-              size = vgpu_size;
-              internal_config = internal_config;
-              identifier = GVT_g conf.identifier;
-              experimental = conf.experimental;
-            })
-    |> collect []
+    List.filter_map (V.vgpu_type_of_conf vendor_name device) whitelist
```

Comment:
(Comment copied from yesterday...)
Rather than using `filter_map` and thus increasing our dependence on the "transitional" `Stdext` library (see comment on `open Listext` etc. at the top of the file), how about this approach to removing the need for the `collect` function?
```
List.map (V.vgpu_type_of_conf vendor_name device) whitelist
|> List.filter (fun x -> x <> None)
```